File: java/03-bootstrap/src/test/java/com/fResult/bootstrap/customers/services/TransactionalCustomerServiceTest.java
package com.fResult.bootstrap.customers.services;

import com.fResult.bootstrap.BootstrapApplication;
import com.fResult.bootstrap.TransactionTestMixin;
import org.junit.jupiter.api.Test;

class TransactionalCustomerServiceTest extends ApplicationContextAwareBaseClass
    implements TransactionTestMixin {

  @Test
  @Override
  public void insert() {
    super.insert();

    testTransactionalityOfSave(getCustomerService());
  }

  @Override
  protected Class<BootstrapApplication> getConfigurationClass() {
    return BootstrapApplication.class;
  }
}

File: kotlin/03-bootstrap/src/test/kotlin/com/fResult/bootstrap/customers/services/TransactionalCustomerServiceTest.kt
package com.fResult.bootstrap.customers.services

import com.fResult.bootstrap.BootstrapApplication
import com.fResult.bootstrap.TransactionTestMixin
import org.junit.jupiter.api.Test
import kotlin.reflect.KClass

class TransactionalCustomerServiceTest : ApplicationContextAwareBaseClass(), TransactionTestMixin {
  @Test
  override fun `should insert customer`() {
    super.`should insert customer`()

    testTransactionalityOfSave(getCustomerService())
  }

  override fun getConfigurationClass(): KClass<BootstrapApplication> = BootstrapApplication::class
}
